openflow13/nx_action.go

```diff
@@ -1336,7 +1336,7 @@ func (a *NXActionNote) UnmarshalBinary(data []byte) error {
 		return errors.New("the []byte is too short to unmarshal a full NXActionNote message")
 	}
 	n := a.NXActionHeader.Len()
-	a.Note = data[n:]
+	a.Note = data[n:a.Length]
```

Comment:
Does this mean that the passed data has trailing stale bytes?

Reply:
> Does this mean that the passed data has trailing stale bytes?

Correct, since the function Instruction.UnmarshalBinary has passed all data to decode actions (https://github.com/contiv/libOpenflow/blob/master/openflow13/instruction.go#L207), it might read the bytes in the subsequent actions which leads to a mistake. So it would be secure to limit the bytes according to the length of the Note action.